aten/src/ATen/native/LossCTC.cpp

```diff
@@ -36,13 +36,14 @@ static inline int64_t get_target_prime(target_t* target, int64_t offset, int64_t
 // The function returns the loss and the alphas, the alphas are kept for the backward step. The wrapper (ctc_loss below) hides
 // the alphas from the user by only returning the loss.
 template<typename scalar_t, ScalarType target_scalar_type>
-std::tuple<Tensor, Tensor> ctc_loss_cpu_template(const Tensor& log_probs, const Tensor& targets, IntArrayRef input_lengths, IntArrayRef target_lengths, int64_t BLANK) {
+std::tuple<Tensor, Tensor> ctc_loss_cpu_template(const Tensor& log_probs_, const Tensor& targets, IntArrayRef input_lengths, IntArrayRef target_lengths, int64_t BLANK) {
   // log_probs: input_len x batch_size x num_labels
   // targets [int64]: batch_size x target_length OR sum(target_lengths)
   constexpr scalar_t neginf = -std::numeric_limits<scalar_t>::infinity();
   using target_t = typename std::conditional<target_scalar_type == kInt, int, int64_t>::type;
 
   CheckedFrom c = "ctc_loss_cpu";
+  Tensor log_probs = log_probs_.dim() == 3 ? log_probs_ : log_probs_.unsqueeze(1);
```

Comment:
I keep getting confused by these unsqueezes without corresponding squeezes :p

Proposal: handle batching at the highest level [`ctc_loss` function](https://github.com/pytorch/pytorch/blob/b15212c62bb00b1153f35e78b8eaa022f9dd867f/aten/src/ATen/native/LossCTC.cpp#L369-L405). There should be no need to touch the CPU / GPU specific impls if you do this. Wdyt?

Reply:
Also works! Probably a bit cleaner, let me do that :)